src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -57,13 +132,152 @@ public boolean equals(Object other) {
         }
 
         DeleteCommand otherDeleteCommand = (DeleteCommand) other;
-        return targetIndex.equals(otherDeleteCommand.targetIndex);
+
+        if (name != null && !name.equals(otherDeleteCommand.name)) {
+            return false;
+        }
+
+        if (nric != null && !nric.equals(otherDeleteCommand.nric)) {
+            return false;
+        }
+
+        return deletePersonDescriptor.equals(otherDeleteCommand.deletePersonDescriptor);
     }
 
     @Override
     public String toString() {
         return new ToStringBuilder(this)
-                .add("targetIndex", targetIndex)
+                .add("nric", nric)
+                .add("name", name)
+                .add("deletePersonDescriptor", deletePersonDescriptor)
                 .toString();
     }
+
+    /**
+     * Creates and returns a {@code Person} with the details of {@code personToEdit}
+     * edited with {@code deletePersonDescriptor}.
+     */
+    public static Person createEditedPerson(Person personToEdit, DeletePersonDescriptor deletePersonDescriptor) {
+        assert personToEdit != null;
+
+        Name updatedName = personToEdit.getName();
+        Nric updatedNric = personToEdit.getNric();
+        Phone updatedPhone = personToEdit.getPhone();
+        Email updatedEmail = personToEdit.getEmail();
+        Address updatedAddress = personToEdit.getAddress();
+        Set<Tag> updatedTags = personToEdit.getTags();
+        Set<MedicalHistory> updatedMedicalHistories = personToEdit.getMedicalHistories();
+        Appointment updatedAppointment = personToEdit.getAppointment().isPresent() ? personToEdit.getAppointment().get()
+                : null;
+
+        if (deletePersonDescriptor.getAppointment()) {
+            updatedAppointment = new Appointment("");
+        }
+
+        return new Person(updatedName, updatedNric, updatedPhone, updatedEmail, updatedAddress, updatedAppointment,
+                updatedMedicalHistories, updatedTags);
+    }
+
+    /**
+     * Stores the boolean value of the fields that is to be deleted from a person in
+     * the address book.
+     */
+    public static class DeletePersonDescriptor {
```

Comment:
This class is functional for now, but perhaps we can make it more in line with `EditPersonDescriptor` in the future?